cmd/oras/manifest/fetchConfig.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manifest
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/spf13/cobra"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content"
+	"oras.land/oras-go/v2/content/oci"
+	"oras.land/oras/cmd/oras/internal/descriptor"
+	oerrors "oras.land/oras/cmd/oras/internal/errors"
+	"oras.land/oras/cmd/oras/internal/option"
+	"oras.land/oras/internal/cache"
+)
+
+type fetchConfigOptions struct {
+	option.Common
+	option.Descriptor
+	option.Pretty
+	option.Remote
+
+	cacheRoot  string
+	mediaType  string
+	outputPath string
+	targetRef  string
+}
+
+func fetchConfigCmd() *cobra.Command {
+	var opts fetchConfigOptions
+	cmd := &cobra.Command{
+		Use:   "fetch-config [flag] name<:tag|@digest>",
+		Short: "[Preview] Fetch the config of a manifest from a remote registry",
+		Long: `[Preview] Fetch the config of a manifest from a remote registry
+
+** This command is in preview and under development. **
+
+Example - Fetch the config:
+  oras manifest fetch-config localhost:5000/hello:latest
+
+Example - Fetch the config and save it to a local file:
+  oras manifest fetch-config --output config.json localhost:5000/hello:latest
+
+Example - Fetch the descriptor of the config:
+oras manifest fetch-config --descriptor localhost:5000/hello:latest
+`,
+		Args: cobra.ExactArgs(1),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if opts.outputPath == "-" && opts.OutputDescriptor {
+				return errors.New("`--output -` cannot be used with `--descriptor` at the same time")
+			}
+
+			opts.cacheRoot = os.Getenv("ORAS_CACHE")
+			return opts.ReadPassword()
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.targetRef = args[0]
+			return fetchConfig(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.outputPath, "output", "o", "", "output file path")
+	cmd.Flags().StringVarP(&opts.mediaType, "media-type", "", "", "media type of the manifest config")
```

Comment:
Can we remove this flag as it does not have a user scenario and it confuses users? /cc @FeynmanZhou @yizha1

Reply:
> Can we remove this flag as it does not have a user scenario and it confuses users? /cc @FeynmanZhou @yizha1

I was tried to support the same feature as [`oras pull --config`](https://github.com/oras-project/oras/blob/main/cmd/oras/pull.go#L119), but if users always use this command for an image manifest, `--media-type` is not required, because we can always recognize the config for an image manifest.

Reply:
Removed `--media-type` for now. I think if only image manifest type support this `manifest fetch-config` command, this argument is unnecessary.